frontend/src/routes/login/logic.ts

```diff
@@ -0,0 +1,28 @@
+import { zodResolver } from '@hookform/resolvers/zod';
+import { z } from 'zod';
+import { useForm } from 'react-hook-form';
+
+export const loginFormSchema = z.object({
+  username: z.string().min(1),
+  password: z.string().min(1),
+});
+
+type ILoginFormSchema = z.infer<typeof loginFormSchema>;
+
+const onSubmit = (_formData: ILoginFormSchema) => {
```

Comment:
Can integrate with login endpoint now

Reply:
E2E to be done once full user service is completed, including signing up endpoints as well as username validation.